src/CoreShop/Component/Order/Cart/CartContextResolverInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+/**
+ * CoreShop.
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2015-2020 Dominik Pfaffenbauer (https://www.pfaffenbauer.at)
+ * @license    https://www.coreshop.org/license     GNU General Public License version 3 (GPLv3)
+ */
+
+namespace CoreShop\Component\Cart\Cart;
+
+use CoreShop\Component\Order\Model\CartInterface;
+
+interface CartContextResolverInterface
+{
+    public function resolveCartContext(CartInterface $cart);
```

Comment:
Other methods that use the context this method returns expect it to be an `array`, so the signature should be:

```suggestion
    public function resolveCartContext(CartInterface $cart): array;
```

Reply:
True, but we currently don't have return-types at all in CoreShop. That will change with CoreShop 3.0. Until that, we don't introduce new ones neither.

Reply:
Oh, okay... in that case there should be some PhpDoc at least, I think.

```suggestion
    /**
     * @return array
     */
    public function resolveCartContext(CartInterface $cart);
```

Reply:
done